selfdrive/controls/lib/pathplanner.py

```diff
@@ -53,41 +56,48 @@ def update(self, CP, VM, CS, md, live100):
     angle_offset = live100.live100.angleOffset
     self.MP.update(v_ego, md)
 
-    # Run MPC
-    self.angle_steers_des_prev = self.angle_steers_des_mpc
+    cur_time = sec_since_boot()
+    self.angle_steers_des_prev = np.interp(cur_time, self.mpc_times, self.mpc_angles)
+
     curvature_factor = VM.curvature_factor(v_ego)
 
     l_poly = libmpc_py.ffi.new("double[4]", list(self.MP.l_poly))
     r_poly = libmpc_py.ffi.new("double[4]", list(self.MP.r_poly))
     p_poly = libmpc_py.ffi.new("double[4]", list(self.MP.p_poly))
 
-    # account for actuation delay
-    self.cur_state = calc_states_after_delay(self.cur_state, v_ego, angle_steers, curvature_factor, CP.steerRatio, CP.steerActuatorDelay)
-
-    v_ego_mpc = max(v_ego, 5.0)  # avoid mpc roughness due to low speed
-    self.libmpc.run_mpc(self.cur_state, self.mpc_solution,
-                        l_poly, r_poly, p_poly,
-                        self.MP.l_prob, self.MP.r_prob, self.MP.p_prob, curvature_factor, v_ego_mpc, self.MP.lane_width)
+    # account for actuation delay AND interpolation
+    self.cur_state = calc_states_after_delay(self.cur_state, v_ego, angle_steers, curvature_factor, CP.steerRatio, CP.steerActuatorDelay + _DT_MPC)
```

Comment:
mmm... why _DT should be added to the delay?

Reply:
That was my initial approach to compensate for the delay caused by interpolation.  In the sketch below, the top image shows how interpolating from previous angle to the new angle creates a delay of 0.05 seconds.  However, I no longer think that the approach above is the correct way to compensate.  It seems that the ideal offset would be 0.025seconds, so that the requested angle passes through the center of the MPC "steps".

If you have a preferred approach to create that offset, please let me know.  It seems that the simplest approach may be to cut the first time increment in half, like this:
```python
   self.mpc_times = [cur_time,
                     cur_time + 0.025,
                     cur_time + 0.075]
```

![image](https://user-images.githubusercontent.com/6308011/57786942-118f0d00-76fa-11e9-8bde-327a4beb86b8.png)